### PR TITLE
Adding release scripts

### PR DESCRIPTION
**Pull request by barney-s**
Adding release targets in makefile.
Simplifying the generation part.

The goal is the aio file would be unique for each release branch.

### CODE REVIEW

**Comment by barney-s** [comment]
/assign @ant31 

Please review @nan-yu

**Comment by barney-s** [comment]
For #166

**Comment by barney-s** [comment]
@ant31 - Would you be able to review and LGTM this ?
@mortent - This is required to generate an official release branch. Would you please review this ?

Also the current version is being pinned at an arbitrary 0.8.1
Any suggestions ?

**Comment by ant31** [comment]
I would tend to prioritize end-user experience and make 'deploy' target not directly using the pre-generated aio file (use default VERSION file too). It removes the need to download all binaries and development dependencies. 
Maintainers would take extra steps anyway and dive more in the configuration

Is the aoi file published in github releases?

File: Makefile [hunk 1]
```diff
@@ -183,14 +188,15 @@ misspell-fix: $(TOOLBIN)/misspell
 ## --------------------------------------
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
+# This is expected to be used by user during dev
 .PHONY: deploy
-deploy: $(TOOLBIN)/kubectl
-	$(TOOLBIN)/kubectl apply -f deploy/kube-app-manager-aio_$(VERSION).yaml
+deploy: $(TOOLBIN)/kubectl generate-resources
```
Comment (ant31) [inline Makefile]: what about a double target ? 
```
deploy-dev
deploy
```
I'd prefer to skip build dependencies and default to the 'release' file for the `make deploy`

Comment (barney-s) [reply Makefile]: ack. will fix it.

File: Makefile [hunk 2]
```diff
@@ -292,3 +296,34 @@ clean:
...
+	git checkout master
+	git rebase upstream/master
+	git branch -D $(RELEASE_BRANCH) || true
+	git checkout -b $(RELEASE_BRANCH)
+	git push -f $(RELEASE_REMOTE) $(RELEASE_BRANCH)
+
+.PHONY: release-tag
+release-tag: release-branch
+	git branch -D $(RELEASE_BRANCH) || true
+	git branch ${RELEASE_BRANCH} ${RELEASE_REMOTE}/${RELEASE_BRANCH}
```
Comment (nan-yu) [inline Makefile:321]: It's confusing that we create the $(RELEASE_BRANCH) branch in the `create-release-branch`, and then delete the branch and fetch again from ${RELEASE_REMOTE}. Why are we doing that?

File: docs/release.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+#### Version files
+For official releases [`VERSION`] file is used.
+For developers [`VERSION-DEV`] file is used.
+
+The default file used is [`VERSION-DEV`].
+To use [`VERSION`], set the `VERSION_FILE` env variable.
+
+Developers should edit the [`VERSION-DEV`] file to set their choice of container registry and version.
+
+## Release procedure
```
Comment (nan-yu) [inline docs/release.md:28]: It's not clear to me which task I should run in order to make a release.
Should I run `make release-branch` first, and then `make release-tag`.
- If so, `release-tag` depends on `release-branch`, run `make release-tag` alone should be okay.
- If not, then `make release-branch` alone doesn't generate any release tag. What do we do with a dangled release branch?

File: docs/release.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+VERSION_FILE=VERSION make release-branch
+
+# Developers use this for their fork
+make release-branch
+```
+
+#### Create release tag
+
+Patch releases are created from the patch branch. 
+Ensure that all necessary fixes are merged, documentation updated and most importantly the `pathc` version is updated in the [`VERSION`] file.
```
Comment (nan-yu) [inline docs/release.md:47]: typo: patch

File: Makefile (hunk 2 above)
Comment (barney-s) [reply Makefile:321]: I see the confusion. We cleanup the local ${RELEASE_BRANCH} and create a release branch pointing to the remote.

File: docs/release.md (hunk 3 above)
Comment (barney-s) [reply docs/release.md:28]: Release branch is usually created once per release $major.$minor where as release tag is major.minor.patch 
So for the first time release branch would be created. 

Calling release-branch explicitly is a good practice.

File: Makefile (hunk 1 above)
Comment (barney-s) [reply Makefile]: Done. Separate make targets added.

**Comment by barney-s** [comment]
> Is the aoi file published in github releases?

Yes.
deploy/kube-app-manager-aio.yaml
for each branch/tag

**Pull request merged**